src/output/plugins/RecorderOutputPlugin.cxx

```diff
@@ -251,14 +249,13 @@ RecorderOutput::ReopenFormat(AllocatedPath &&new_path)
 	assert(path.IsNull());
 	assert(file == nullptr);
 
-	auto *new_file = new FileOutputStream(new_path);
+	auto new_file = std::make_unique<FileOutputStream>(new_path);
 
 	AudioFormat new_audio_format = effective_audio_format;
 
 	try {
 		encoder = prepared_encoder->Open(new_audio_format);
 	} catch (...) {
-		delete new_file;
```

Comment:
Being able to avoid these would be nice, but not at the cost of adding pointless overhead to `~RecorderOutput()`. Maybe call `file = new_file.release()` later in this method?